Tests/web3swiftTests/localTests/ST20AndSecurityTokenTests.swift

```diff
@@ -0,0 +1,141 @@
+            guard let function = self.st20token.contract.contract.getFunctionCalled(transaction.data) else {
+                XCTFail("Failed to decode function call to determine what shall be returned")
+                return Data()
+            }
+            switch function.name {
+            case "balanceOf":
+                let address = function.decodeInputData(transaction.data)?["0"] as? EthereumAddress
+                XCTAssertEqual(address, userAddress)
+                return ABIEncoder.encode(types: [.uint(bits: 256)], values: [expectedBalance] as [AnyObject])!
+            case "allowance":
+                let transactionInput = function.decodeInputData(transaction.data)
+                XCTAssertEqual(transactionInput?["0"] as? EthereumAddress, userAddress)
+                XCTAssertEqual(transactionInput?["1"] as? EthereumAddress, delegate)
+                return ABIEncoder.encode(types: [.uint(bits: 256)], values: [expectedAllowance] as [AnyObject])!
+            default:
+                // Unexpected function called
+                XCTFail("Called function '\(String(describing: function.name))' which wasn't supposed to be called.")
+                return Data()
+            }
+        }
+
+        let balance = try await st20token.getBalance(account: userAddress)
+        let allowance = try await st20token.getAllowance(originalOwner: userAddress, delegate: delegate)
+        XCTAssertEqual(balance, expectedBalance)
+        XCTAssertEqual(allowance, expectedAllowance)
+    }
+
+    func testSecurityTokenInvestors() async throws {
+        let expectedNumberOfInvestors = BigUInt.randomInteger(lessThan: BigUInt(10000000000))
+//        let expectedInvestors = [EthereumAddress("0xe22b8979739D724343bd002F9f432F5990879901")!,EthereumAddress("0x2dD33957C90880bE4Ee9fd5F703110BDA2E579EC")!]
+
+        ethMock.onCallTransaction = { transaction in
+            guard let function = self.securityToken.contract.contract.getFunctionCalled(transaction.data) else {
+                XCTFail("Failed to decode function call to determine what shall be returned")
+                return Data()
+            }
+            switch function.name {
+            case "investorCount":
+                return ABIEncoder.encode(types: [.uint(bits: 256)], values: [expectedNumberOfInvestors] as [AnyObject])!
+//            case "investors":
+//                return ABIEncoder.encode(types: [.array(type: .address, length: 0)],
+//                                         values: [expectedInvestors] as [AnyObject])!
+            default:
+                // Unexpected function called
+                XCTFail("Called function '\(String(describing: function.name))' which wasn't supposed to be called.")
+                return Data()
+            }
+        }
+
+        let investorsCount = try await securityToken.investorCount()
+        XCTAssertEqual(investorsCount, expectedNumberOfInvestors)
+//        XCTAssertEqual(investors, expectedInvestors)
```

Comment:
Comments should be dropped though.

Reply:
It's definitely not coming with this PR, just wanted to highlight that we have issues with either ABI or the implementation.
One of the conclusions that I have is that each and every ABI like `Web3.Utils.st20ABI` and the rest must be precisely the version of ABI that matches the original standard without any deviation. 
Since we allow developers to reference these ABIs from our project it's our responsibility to:
- validate the ABI (it must match the standard);
- give the link to the place where we got this ABI from;
- recommend developers to use their own ABIs if that's possible as they will have more control over that.

Reply:
@yaroslavyaroslav @janndriessen Here is the fix for these commented out lines - https://github.com/web3swift-team/web3swift/pull/704

The branch with the fix is not based on the one in the current PR so I'll be able to update tests only when the fix is merged into `develop`.